tests/integration/page_elements.py

```diff
@@ -0,0 +1,324 @@
+from lazy.lazy import lazy
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.support.select import Select
+from group_project_v2.components.stage import StageState
+
+__author__ = 'e.kolpakov'
+
+
+class BaseElement(object):
+    def __init__(self, browser, element):
+        self._browser = browser
+        self._element = element
+
+    @property
+    def browser(self):
+        return self._browser
+
+    @property
+    def element(self):
+        return self._element
+
+    def __getattr__(self, item):
+        if hasattr(self.__dict__, item):
+            return getattr(self.__dict__, item)
+        if hasattr(self.element, item):
+            return getattr(self.element, item)
+        return super(BaseElement, self).__getattr__(item)
+
+    def make_element(self, dom_element, element_type):
+        return element_type(self.browser, dom_element)
+
+
+    def make_elements(self, css_selector, element_type):
+        elements = self.element.find_elements_by_css_selector(css_selector)
```

Comment:
@Kelketek good point. Actually I've decided not to use PageObjects as it just get in my way instead of helping: basically, most elements here can't be identified by css_selector (as there are many of the same type having the same css classes), or need some significant boilerplate to target (i.e. determining activity XBlock id to match by `[data-activity-id='']` query). Instead, I've taken the idea of isolating tests from page interaction and created a minimal thing that suited my needs while maintaining that isolation.

So, this is basically a lightweight PageObject modified to operate under the following constraints:
* There are no url to visit.
* There can be multiple instances of the same type of object on the same page.
* is_browser_on_page check is redundant as majority of tests run on a single page; basically `is_browser_on_page` check is replaced with `NoSuchElementException` throwed by Selenium when corresponding DOM element can't be found.

Also, note that `PageObject.q`  returns a collection of generic PageObjects (or even just thin wrappers around selenium WebElement), whereas this method allows specifying concrete wrapper type.